server/model/monitor.js

```diff
@@ -527,19 +539,22 @@ class Monitor extends BeanModel {
                     bean.msg = `${res.status} - ${res.statusText}`;
                     bean.ping = dayjs().valueOf() - startTime;
 
-                    // Store certificate and check for expiry if https is used
+                    // fallback for if kelog event is not emitted, but we may still have tlsInfo,
+                    // e.g. if the connection is made through a proxy
                     if (this.getUrl()?.protocol === "https:") {
-                        // No way to listen for the `secureConnection` event, so we do it here
-                        const tlssocket = res.request.res.socket;
-
-                        if (tlssocket) {
-                            tlsInfo.valid = tlssocket.authorized || false;
-                        }
-
-                        await this.updateTlsInfo(tlsInfo);
-                        if (!this.getIgnoreTls() && this.isEnabledExpiryNotification()) {
-                            log.debug("monitor", `[${this.name}] call checkCertExpiryNotifications`);
-                            await this.checkCertExpiryNotifications(tlsInfo);
+                        if (tlsInfo.valid === undefined) {
```

Comment:
nit: What do you think?
should we merge these two if statements into one line? (removing the one above which I can't comment on)

```suggestion
                    if (this.getUrl()?.protocol === "https:" && tlsInfo.valid === undefined) {
```